pyxem/signals/polar_diffraction2d.py

```diff
@@ -130,6 +130,51 @@ def get_angular_power(self, mask=None, normalize=True, inplace=False, **kwargs):
         fourier_axis.scale = 1
         return power
 
+    def get_pearson_correlation(self, mask=None, kmin=None, kmax=None, inplace=False, **kwargs):
+        """Returns the pearson rotational correlation in the form of a Signal2D class.
+
+        Parameters
+        ----------
+        mask: Numpy array
+            A bool mask of values to ignore of shape equal to the signal shape.
+        kmin: float
+            minimum k value in corresponding unit for segment correlation (None if use
+            the entire pattern)
+        kmax: float
+            maximum k value in corresponding unit for segment correlation (None if use
+            the entire pattern)
+        inplace: bool
+            From hyperspy.signal.map(). inplace=True means the signal is
+            overwritten.
+
+        Returns
+        -------
+        correlation: Signal2D
+        """
+        if kmin is not None:
```

Comment:
what happens if kmax is None? I think you're missing some logic in here, and it would be more natural to read if the short statement was at top, ie:

```python
if kmin is None and kmax is None:
      correlation = self._map_iterate(_pearson_correlation, mask=mask, inplace=inplace, **kwargs)
else:
     .....
```

Reply:
I've thought about this. The problem is kmin and kmax has to be both None or both assigned in order for this method to work. Should I merge them together as one tuple argument say `krange`, and set its default as None? That might be better.

Reply:
That would work, alternatively, make the documentation very clear that kmin and kmax must either both be set, or neither of them should be set. Then:

```python
if kmin is None and kmax is None:
       # this does both are none case
elif kmin is None or kmax is None:
       # this catches only one is none, can't be both because the block above deals with that
       raise ValueError('Either both or neither of kmin and kmax must be set to None for this method')
else:
      # kmin and kmax can't be None now, so we can run with them both as floats
```